py/src/braintrust/framework.py

```diff
@@ -445,31 +445,134 @@ def _make_eval_name(name: str, experiment_name: Optional[str]):
     return out
 
 
-def Eval(
+def _EvalCommon(
     name: str,
-    data: Callable[[], Union[Iterator[EvalCase], AsyncIterator[EvalCase]]],
+    data: EvalData,
     task: Callable[[Input, EvalHooks], Union[Output, Awaitable[Output]]],
     scores: List[EvalScorer],
     experiment_name: Optional[str] = None,
     trial_count: int = 1,
     metadata: Optional[Metadata] = None,
     is_public: bool = False,
     update: bool = False,
-    reporter: Optional[Union[ReporterDef, str]] = None,
+    reporter: Optional[ReporterDef] = None,
```

Comment:
Can we avoid the argument defaults if everything is being delegated to this method? Just thinking of ways to reduce the amount of argument copy-paste. Seems like there's a `TypedDict` and an `Unpack` thing.

Reply:
Unfortunately `TypedDict` + `Unpack` breaks the doc generator, so we can't use it on public API:

![Screenshot 2024-11-12 at 11 27 49 AM](https://github.com/user-attachments/assets/583bd4db-0452-46be-aa40-4879aefd922e)

This leaves the internal `_EvalCommon`. We can move the argument to `TypedDict`, but this is not reducing repetition, while making the usage less convenient (need to type `kwargs["arg_name"]`).

I will settle for removing the argument defaults.